src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -139,35 +140,54 @@ internal PEReader PEReaderOpt
             }
         }
 
-        internal unsafe MetadataReader MetadataReader
+        internal MetadataReader MetadataReader
         {
             get
             {
                 if (_lazyMetadataReader == null)
                 {
-                    MetadataReader newReader;
+                    InitializeMetadataReader();
+                }
+                else if (_isDisposed)
```

Comment:
No the pattern is:
1) host creates ModuleMetadata 
2) hosted code accesses PE Symbols from a compilation provided by the host 
3) host disposes ModuleMetadata because it thinks nobody is using the PE Symbols anymore
4) hosted code uses the PE Symbols